app/models/article.rb

```diff
@@ -72,6 +72,9 @@ class Article < ApplicationRecord
   validate :validate_collection_permission
   validate :validate_tag
   validate :validate_video
+  validate :validate_co_authors
+  validate :validate_co_authors_must_not_be_the_same
+  validate :validate_co_authors_exist
```

Comment:
Idea: all of these validations check `if co_author_ids.blank?` as the very first thing within the validation method. If you want to tidy up these validations _even further_, you could extract that out into a [method](https://edgeguides.rubyonrails.org/active_record_validations.html#using-a-symbol-with-if-and-unless), or even better, a [proc](https://edgeguides.rubyonrails.org/active_record_validations.html#using-a-proc-with-if-and-unless), and then remove that repeated check from all three of these. Something like this:

```ruby
validates : validate_co_authors, unless: -> { co_author_ids.blank? }
validates : validate_co_authors_must_not_be_the_same, unless: -> { co_author_ids.blank? }
validates : validate_co_authors_exist, unless: -> { co_author_ids.blank? }
```